### PR TITLE
Fix CHECK() when split view is opened while there are pinned tabs

### PR DESCRIPTION
**Pull request by sangwoo108**
* IsValidModelIndex() check was wrong, because the implementation of the concrete class, TabContainerImpl, was using the view index not the model index.

Resolves https://github.com/brave/brave-browser/issues/37564

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/views/tabs/brave_tab_container.cc [hunk 1]
```diff
@@ -297,11 +297,29 @@ void BraveTabContainer::PaintBoundingBoxForTiles(
...
     gfx::Canvas& canvas,
     const SplitViewBrowserData::Tile& tile) {
+  if (!GetTabCount()) {
+    return;
+  }
+
+  // Note that GetTabAtModelIndex() and IsValidModelIndex() is actually using
+  // "ViewModel" index. See TabContainerImpl::GetTabAtModelIndex(), and
+  // implementations in compound_tab_container.cc implementation. Thus, we need
+  // to add pinned tab count. Also NumPinnedTabsInModel() returns 0 when
```
Comment (simonhong) [inline browser/ui/views/tabs/brave_tab_container.cc:307]:
```suggestion
  // to add pinned tab count. Also NumPinnedTabsInModel() returns 0 when ?
```

File: browser/ui/views/tabs/brave_tab_container.cc [hunk 2]
```diff
@@ -297,11 +297,29 @@ void BraveTabContainer::PaintBoundingBoxForTiles(
...
+  const int offset =
+      is_pinned_tab_container ? 0 : tab_strip_model->IndexOfFirstNonPinnedTab();
+
+  auto tab1_index = tab_strip_model->GetIndexOfTab(tile.first) - offset;
+  auto tab2_index = tab_strip_model->GetIndexOfTab(tile.second) - offset;
+  if (!controller_->IsValidModelIndex(tab1_index) ||
+      !controller_->IsValidModelIndex(tab2_index)) {
+    // In case the tiled tab is not in this container, this can happen.
+    // For instance, this container is for pinned tabs but the tile is
+    // unpinned.
```
Comment (simonhong) [inline browser/ui/views/tabs/brave_tab_container.cc:320]: In this comment, `tile` is some what confusing.
IIUC, `tile` is a set of tabs for split view. (Am I right?)
If it's right, there is one pinned tab and unpinned tab in curren tile in this case.
`tile is unpinned` makes me confusing.

Comment (sangwoo108) [reply browser/ui/views/tabs/brave_tab_container.cc:320]: Yes, `tabs in the tile is unpinned` would be better. The reason why I named it `tile` is that we want to have room for having multiple tab tiling, like 3+ tabs in a view.

File: browser/ui/views/tabs/brave_tab_container.cc (hunk 1 above)
Comment (sangwoo108) [reply browser/ui/views/tabs/brave_tab_container.cc:307]: The line can be removed, as we don't use TabContainerImpl::NumPinnedTabs() anymore. Thanks for spotting.

**Review by simonhong — APPROVED** [review]
++

**Pull request merged**